api/types/accesslist/accesslist.go

```diff
@@ -157,11 +171,48 @@ func (a *AccessList) CheckAndSetDefaults() error {
 		return trace.BadParameter("owners are missing")
 	}
 
-	if a.Spec.Audit.Frequency == 0 {
-		return trace.BadParameter("audit frequency must be greater than 0")
+	// This interval is the number of months in between reviews. This will only be used if the
+	// recurrence string is not set.
+	recurrenceInterval := 6
+
+	// We know that the audit field needs a recurrence conversion if the frequency is set.
+	if a.Spec.Audit.Frequency != 0 {
+		// We'll modify the recurrence interval based on the value of the frequency. We'll take the assumptions made in the
+		// UI and max out at a yearly recurrence cycle
+		recurrenceInterval = int(a.Spec.Audit.Frequency / estimatedHoursInAMonth)
+		if recurrenceInterval > 12 {
+			recurrenceInterval = 12
+		} else if recurrenceInterval < 1 {
+			recurrenceInterval = 1
+		}
+
+		// Unset the frequency.
+		a.Spec.Audit.Frequency = 0
+
+		// If the audit field needs a conversion to the new recurrence format, we know that that the next audit date may not
+		// be populated. This will not happen going forward, as we now check for the zero value of the next audit date.
+		// We'll ensure that the next audit date is set to a sane default before proceeding.
+		if a.Spec.Audit.NextAuditDate.IsZero() {
+			a.Spec.Audit.NextAuditDate = time.Date(2024, 1, 1, 0, 0, 0, 0, time.UTC)
+		}
```

Comment:
So the logic is that it shouldn't be realistically possible to trigger this logic past this date, but I don't disagree with you. I can potentially try to do a `time.Now()` thing here instead of this. Thoughts?